src/broker.rs

```diff
@@ -0,0 +1,77 @@
+use crate::MQTT_INSECURE_DEFAULT_PORT;
+use embedded_nal::{nb, AddrType, Dns, IpAddr, Ipv4Addr, SocketAddr};
+
+pub trait Broker {
```

Comment:
The intent of being generic is that a user may or may not have DNS support. If we make it a known type, then we enforce that `Stack: Dns`. With this API, we allow the type to optionally contain something that `impl Dns` instead of forcing this requirement on the user.

Reply:
Hmm. Ok. It's just a bit noisy with the number of type params.

Maybe another approach to get rid of the type param would be to have the user resolve the broker before connecting, like `set_broker_port()`. The user just may have to call it repeatedly until the lookup is complete...

Reply:
It's essentially a question on usability. Not sure if it's more ergonomic to force the user to resolve things or to just add one more type param

Reply:
I guess another option would have been to remove the nal dns trait and just error from the stack if it doesn't support DNS.

(ref trait discussion here: https://github.com/rust-embedded-community/embedded-nal/issues/35).

Let's do the type param for now and maybe keep in mind possible nal redesign and v1 paths. (embedded-hal is also inching towards v1 and having the traits all "stable" sounds like something that could be desirable).

Reply:
Sounds good. This is ready to go whenever then